backend/src/modules/doubles-tournament/core/domain/QueuedMatch.ts

```diff
@@ -0,0 +1,70 @@
+import { TeamId } from './TeamId';
+import { TournamentId } from './TournamentId';
+import { DoublesTournament } from './DoublesTournament';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+import { MatchesQueue } from './MatchesQueue';
+import { MatchWasQueued } from './event/MatchWasQueued';
+
+export class QueuedMatch {
+  readonly matchNumber: number;
+  readonly team1Id: TeamId;
+  readonly team2Id: TeamId;
+
+  constructor(props: { matchNumber: number; team1Id: TeamId; team2Id: TeamId }) {
+    this.matchNumber = props.matchNumber;
+    this.team1Id = props.team1Id;
+    this.team2Id = props.team2Id;
+  }
+}
+
+export function pushMatchToQueue(
+  tournament: DoublesTournament | undefined,
+  queue: MatchesQueue | undefined,
+  command: {
+    tournamentId: TournamentId;
+    matchNumber: number;
+    team1Id: TeamId;
+    team2Id: TeamId;
+  },
+  currentTime: Date,
+): DomainCommandResult<MatchesQueue> {
+  if (tournament === undefined) {
+    throw new Error("This tournament doesn't exists.");
+  }
+  if (command.matchNumber <= 0) {
+    throw new Error('Such match number is incorrect!');
+  }
+  if (queue === undefined) {
+    queue = new MatchesQueue({
+      tournamentId: command.tournamentId,
+      queuedMatches: [],
+    });
+  }
+  if (isMatchAlreadyInQueue(command.matchNumber, queue)) {
+    throw new Error('Such match is already waiting in matches queue!');
+  }
+
+  const matchToPush = new QueuedMatch({
+    matchNumber: command.matchNumber,
+    team1Id: command.team1Id,
+    team2Id: command.team2Id,
+  });
+
+  queue.queuedMatches.push(matchToPush);
```

Comment:
Tutaj za dużo pokazujesz, tzn. bierzesz queue.queuedMatches a wg. zasady "Law of Demeter", albo lepiej posłuchać tego: 
https://youtu.be/cJDDsSj2vJA?t=1101

Zobacz tutaj jakoś do 22 :) I czytajcie Pull Requesty innych - nawet te po review! Bo z pewnością wrzucałem już to @PiotrWR kiedyś. 

W swoich obiektach powinieneś zostawiać tylko to co jest możliwe, czyli z defaultu wszystkie pola są private. A już szczególnie lista. Bo obecnie mogę nawet zrobić coś takiego: `queue.queuedMatches.length = 0`.
Czyli musisz zrobić dla MatchesQueue metody, które pokazują co z tym obiektem można zrobić. Na naszej kolejce meczów nie można też filtrować itp. nie chcemy tego :)

Reply:
dobra chyba zrozumiałem :-), poprawię